code/datums/components/waddling.dm

```diff
@@ -21,6 +21,8 @@
 /datum/component/waddling/proc/Waddle()
 	SIGNAL_HANDLER
 
-	animate(parent, pixel_z = 4, time = 0)
-	animate(pixel_z = 0, transform = turn(matrix(), pick(-12, 0, 12)), time=2)
-	animate(pixel_z = 0, transform = matrix(), time = 0)
+	var/rot_degrees = pick(-12, 0, 12)
```

Comment:
What was wrong with previous implementation?

Reply:
The previous implementation always set the transform to the identity matrix when it was done, which overrides size changes and rotations. This method undoes the rotation without incorrect assumptions about the original state.